deepmd/pt/model/task/dipole.py

```diff
@@ -55,13 +90,34 @@ def forward(self, inputs, atype, atype_tebd, rot_mat):
         -------
         - vec_out: output vector. Its shape is [nframes, nloc, 3].
         """
-        nframes, nloc, _ = inputs.size()
-        if atype_tebd is not None:
-            inputs = torch.concat([inputs, atype_tebd], dim=-1)
-        vec_out = self.filter_layers[0](inputs)  # Shape is [nframes, nloc, m1]
-        assert list(vec_out.size()) == [nframes, nloc, self.out_dim]
-        vec_out = vec_out.view(-1, 1, self.out_dim)
-        vec_out = (
-            torch.bmm(vec_out, rot_mat).squeeze(-2).view(nframes, nloc, 3)
-        )  # Shape is [nframes, nloc, 3]
-        return vec_out
+        in_dim = self.dim_descrpt + self.numb_fparam + self.numb_aparam
+
+        nframes, nloc, _ = descriptor.shape
+        gr = gr.view(nframes, nloc, -1, 3)
+        out_dim = gr.shape[2]  # m1
+        self.filter_layers = NetworkCollection(
+            1 if self.distinguish_types else 0,
+            self.ntypes,
+            network_type="fitting_network",
+            networks=[
+                FittingNet(
+                    in_dim,
+                    out_dim,
```

Comment:
@wanghan-iapcm ,the `out_dim` here should be `m1` from `gr`, right? I am having shape mismatch with the mask in GeneralFitting `mask = (atype == type_i).unsqueeze(-1)`.